### DISCUSSION TITLE
Help with acceleration calculations.

### DISCUSSION THREAD

**Discussion by coldsilk** [gin66/FastAccelStepper#139] [General]
I'm brand new to this and I am trying for ~45 degrees per second with a microstepping of 16 and a 1.8 degree motor.

~8.88 = steps_per_degree = ((360.0f / step_angle) * microsteps) / 360.0f;
So I use fas->setSpeedInUs(45 * 8.88);
For acceleration, the header has the example of 10,000 total over 10 seconds (10000 / 10 = 1000).
So I use fas->setAcceleration((45 * 8.88) / 1));

The above is not correct as if I use fas->runForward() the speed is ~4x faster.

Am I supposed to always use move/moveTo and effectively ignore movement with runForward() to make the math work correctly (whatever the correct math is)? Right now I'm imagining acceleration as 20% of the total travel time (10 in front/ 10 in back), however, I don't know how to work the math on that in combination with setAcceleration().

**Comment by gin66** [comment]
Your motor does 200steps/revolution.
so 25steps for 45°. With 16x microstepping: 400 steps for 45°.
If you like to run 45° in 1s, then you need 400steps/s and can use `setSpeedInHz(400);`

For some math check (https://github.com/gin66/FastAccelStepper/blob/master/extras/doc/ramp.md)
